pypika/terms.py

```diff
@@ -288,57 +288,116 @@ def get_sql(self, **kwargs: Any) -> str:
         raise NotImplementedError()
 
 
+def idx_placeholder_gen(idx: int) -> str:
+    return str(idx + 1)
+
+
+def named_placeholder_gen(idx: int) -> str:
+    return f'param{idx + 1}'
+
+
 class Parameter(Term):
     is_aggregate = None
 
-    def __init__(self, placeholder: Union[str, int]) -> None:
+    def __init__(self, placeholder: Union[str, int, Callable[[int], str]] = idx_placeholder_gen) -> None:
         super().__init__()
-        self.placeholder = placeholder
+        self._placeholder = placeholder
+
+    @property
+    def placeholder(self):
+        if callable(self._placeholder):
+            return self._placeholder(None)
```

Comment:
how does this work on None?

Reply:
callable Returns `False`
![image](https://github.com/kayak/pypika/assets/918128/cd63caf9-4075-4f9d-b65b-1d0943b12dc0)

But I presume you mean how will the `self_placeholder(None)` work with None.
Good point, it wouldn't atm with the default generators.

I will fix it now.